File: day1/task1.c
#include <stdio.h>
#include <stdlib.h>

#define LINE_MAX 256

int
main(int argc, char** argv)
{
	if (argc != 2) {
		printf("Please specify input text file as only argument.\n");
		return EXIT_FAILURE;
	}

	FILE* input;
	input = fopen(argv[1], "r");
	if (!input) {
		printf("Not a valid file: %s\n", argv[1]);
		return EXIT_FAILURE;
	}

	char line[LINE_MAX];
	int frequency = 0;
	while (fgets(line, LINE_MAX, input)) {
		frequency += atoi(line);
	}

	printf("Frequency: %d\n", frequency);

	return EXIT_SUCCESS;
}

File: day1/in_go/task1.go
package main

import (
	"fmt"
	"os"
	"bufio"
	"strconv"
)

func die(v ...interface{}) {
	fmt.Println(v...)
	os.Exit(1)
}

func main() {
	if len(os.Args) != 2 {
		die("Please specify input text file as only argument")
	}

	file, err := os.Open(os.Args[1])
	if err != nil {
		die(err)
	}
	defer file.Close()

	frequency := 0

	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		diff, err := strconv.Atoi(scanner.Text())
		if err != nil {
			die(err)
		}
		frequency += diff
	}

	fmt.Println("Freqency:", frequency)
}
